deepmind/open_spiel
# Fixes necessary to upgrade to TF2.2 / Python 3.8 / Ubuntu 20.04

These are some fixes requires to get OpenSpiel compiling and running on Ubuntu 20.04 under TF 2.2 to address https://github.com/deepmind/open_spiel/issues/166. Note that TF2.2 requires Python 3.8.

In particular, this PR changes the following:

- add `tf.disable_v2_behavior` in several places
- upgrade some pip package versions (matplotlib, cvxopt, attrs). 
- remove dependency on dm-sonnet, which does not play well with disable_v2_behavior
- add `simple_nets.py` , a very simple pure TF replacement for sonnet (MLPs and dense layers)

Tested (so far):
- I have run `kuhn_nfsp.py` to ensure the graph looks like what we expect
- Similarly for `kuhn_policy_gradient.py`

Not done yet:
- Apply Deep CFR fix we received earlier today ( https://github.com/deepmind/open_spiel/pull/248 )
- Apply A2C fix we received earlier today ( https://github.com/deepmind/open_spiel/pull/247 )
- Get opinions on changes from people who are more familiar with TF than me :)
